### ISSUE TITLE
Accessibility: Success messages on forms on My Account pages not announced to screen reader users

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43636]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the My Account pages, when submitting forms (edit address, account details), the page refreshes rather than submitting the form with ajax. There is a visual error message, but screen reader users are loaded at the top of the page and do not hear the message.
<img width="1048" alt="Screenshot 2024-01-14 at 10 57 45 PM" src="https://github.com/woocommerce/woocommerce/assets/13153456/48fc1eef-82e7-4b88-95f3-36dc244cb894">

This is likely connected to https://github.com/woocommerce/woocommerce/issues/43635

### Expected behavior

When a form error is present, there should be an auditory message for screen reader users to know that the form submitted successfully.

You can either do this by:
A) When the page reloads, set keyboard focus on the status message so it is read out. OR
B) Add an [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) region on the status message and trigger it after the page loads.

You may also consider changing this form to submit with ajax, so screen reader users are not sent to the top of the page as happens with a refresh.

### Actual behavior

Screen reader users don't easily know if teh form submitted successfully.

### Steps to reproduce

1. Turn on a screen reader such as VoiceOver or NVDA.
2. Go to the My Account > Account details page.
3. Edit something in your account information and submit the form.
5. Listen as the page reloads and observe there's no announcement about the success.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.
